src/BreakpointHooks.js

```diff
@@ -0,0 +1,112 @@
+/* eslint react/prop-types: 0 */
+// @flow
+import * as React from 'react';
+
+function getCurrentViewportSize() {
+  return {
+    width: window.innerWidth,
```

Comment:
Needs a safety check for `window` being undefined. Maybe in server rendering we return `{ width: Infinity, height: Infinity }`, or some large number. I'm not totally sure what the correct behavior there is.